pkg/engine/internal/semconv/identifier.go

```diff
@@ -0,0 +1,173 @@
+)
+
+// NewIdentifier creates a new column identifier from given name, column type, and data type.
+func NewIdentifier(name string, ct types.ColumnType, dt types.DataType) *Identifier {
+	return &Identifier{
+		columnName: name,
+		columnType: ct,
+		dataType:   dt,
+	}
+}
+
+// Identifier is the unique identifier of a column.
+// The uniqueness of a column is defined by its data type, column type (scope), and name.
+// The fully qualified name is represented as [DATA_TYPE].[COLUMN_TYPE].[COLUMN_NAME].
+type Identifier struct {
+	columnName string
+	columnType types.ColumnType
+	dataType   types.DataType
+}
+
+func (i *Identifier) DataType() types.DataType {
+	return i.dataType
+}
+
+func (i *Identifier) ColumnType() types.ColumnType {
+	return i.columnType
+}
+
+func (i *Identifier) Name() string {
+	return i.columnName
+}
+
+func (i *Identifier) Scope() Scope {
+	return ScopeForColumnType(i.columnType)
+}
+
+func (i *Identifier) String() string {
+	return fmt.Sprintf("%s.%s.%s", i.dataType, i.columnType, i.columnName)
+}
+
+// FQN returns the fully qualified name of the identifier.
+func (i *Identifier) FQN() string {
+	return i.String()
+}
+
+func (i *Identifier) Equal(other *Identifier) bool {
+	if i == nil || other == nil {
+		return false
+	}
+	return i.columnName == other.columnName &&
+		i.columnType == other.columnType &&
+		i.dataType == other.dataType
+}
+
+// FQN returns a fully qualified name for a column by given name, column type, and data type.
+func FQN(name string, ct types.ColumnType, dt types.DataType) string {
+	return NewIdentifier(name, ct, dt).FQN()
+}
+
+// ParseFQN returns an [Identifier] from the given fully qualified name.
+// It returns an error if the name cannot be parsed.
+func ParseFQN(fqn string) (*Identifier, error) {
+	if fqn == "" {
+		return nil, errors.New("empty identifier")
+	}
+
+	dt, rest, found := strings.Cut(fqn, ".")
+	if !found {
+		return nil, errors.New("missing data type separator")
+	}
+	if rest == "" {
+		return nil, errors.New("missing column type")
+	}
+
+	dataType, err := types.FromString(dt)
+	if err != nil {
+		return nil, err
+	}
+
+	ct, columnName, found := strings.Cut(rest, ".")
+	if !found {
+		return nil, errors.New("missing column type separator")
+	}
+	if columnName == "" {
+		return nil, errors.New("missing column name")
+	}
+	columnType := types.ColumnTypeFromString(ct)
+
+	return &Identifier{
+		columnName: columnName,
+		columnType: columnType,
+		dataType:   dataType,
+	}, nil
+}
+
+// ParseFQN returns an [Identifier] from the given fully qualified name.
+// It panics if the name cannot be parsed.
+func MustParseFQN(fqn string) *Identifier {
+	ident, err := ParseFQN(fqn)
+	if err != nil {
+		panic(fmt.Sprintf("parsing fqn: %s", err))
+	}
+	return ident
+}
+
+type scopeOrigin string
+type scopeType string
+
+// Scope decribes the origin and type of an identifier.
+type Scope struct {
```

Comment:
i am not sure about the use of `Scope` for naming, this could also be confused with otel scope attributes